### PR TITLE
Remove type annotations from functions.py

### PR DESCRIPTION
**Pull request by tpoterba**
Causing far more trouble than it's worth.

### CODE REVIEW

**Comment by tpoterba** [comment]
I left ones on return types that are unambiguous. I think this is a decent way forward for now.

File: python/hail/expr/functions.py [hunk 1]
```diff
@@ -373,11 +373,7 @@ def chisq(c1: Int32Expression, c2: Int32Expression, c3: Int32Expression, c4: Int
 
 
 @typecheck(c1=expr_int32, c2=expr_int32, c3=expr_int32, c4=expr_int32, min_cell_count=expr_int32)
-def ctt(c1: Int32Expression,
-        c2: Int32Expression,
-        c3: Int32Expression,
-        c4: Int32Expression,
-        min_cell_count: Int32Expression) -> StructExpression:
+def ctt(c1, c2, c3, c4, min_cell_count):
```
Comment (patrick-schultz) [inline python/hail/expr/functions.py]: Should the `-> StructExpression` stay here?

File: python/hail/expr/functions.py [hunk 2]
```diff
@@ -3247,7 +3241,7 @@ def float64(expr: Union[int, float, str, NumericExpression, BooleanExpression, S
         return expr._method("toFloat64", tfloat64)
 
 @typecheck(expr=expr_oneof(expr_numeric, expr_bool, expr_str))
-def float32(expr: Union[int, float, str, NumericExpression, BooleanExpression, StringExpression]) -> Float32Expression:
+def float32(expr: Union[int, float, str, NumericExpression, StringExpression]) -> Float32Expression:
```
Comment (patrick-schultz) [inline python/hail/expr/functions.py]: Why the `Expression` annotations on parameters in conversions only?

**Comment by tpoterba** [comment]
I thought about this while walking to the train, and now I think we should just remove them all. That OK with you?

**Comment by patrick-schultz** [comment]
Yeah, you understand the tradeoffs a lot better than I do.

**Comment by patrick-schultz** [comment]
I still can't figure out the pattern of which annotations you removed.

**Comment by tpoterba** [comment]
I removed all Expression type annotations. I left the others, like HailType and Callable.

**Comment by patrick-schultz** [comment]
Ah, got it.

File: python/hail/expr/functions.py [hunk 3]
```diff
@@ -1443,10 +1440,10 @@ def pnorm(x: Float64Expression) -> Float64Expression:
...
 
 @typecheck(x=expr_float64, lamb=expr_float64, lower_tail=expr_bool, log_p=expr_bool)
-def ppois(x: Float64Expression,
-          lamb: Float64Expression,
-          lower_tail: BooleanExpression = True,
-          log_p: BooleanExpression = False) -> Float64Expression:
+def ppois(x,
+          lamb,
+          lower_tail=True,
+          log_p=False):
```
Comment (patrick-schultz) [inline python/hail/expr/functions.py]: Probably fine all on one line now.

File: python/hail/expr/functions.py [hunk 4]
```diff
@@ -1539,10 +1536,10 @@ def qnorm(p: Float64Expression) -> Float64Expression:
...
 
 @typecheck(p=expr_float64, lamb=expr_float64, lower_tail=expr_bool, log_p=expr_bool)
-def qpois(p: Float64Expression,
-          lamb: Float64Expression,
-          lower_tail: BooleanExpression = True,
-          log_p: BooleanExpression = False) -> Float64Expression:
+def qpois(p,
+          lamb,
+          lower_tail=True,
+          log_p=False):
```
Comment (patrick-schultz) [inline python/hail/expr/functions.py]: Likewise.

**Pull request merged**